conformance/tests/mesh-consumer-route.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: mesh-echo-add-header
+  namespace: gateway-conformance-mesh-consumer
+spec:
+  parentRefs:
+  - kind: Service
+    name: echo-v1
+    namespace: gateway-conformance-mesh
+  rules:
+  - filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: X-Header-Set
+          value: set
+    backendRefs:
+    - name: echo-v1
```

Comment:
did you intend for it to not have `namespace` set? The route is valid, just not what we had before

note the test is only checking `Backend: echo-v1` but we have it in both namespaces, so its untested if we go to the correct namespace?

Reply:
I did intend it to be the same as the omitted backendRefs route, I guess I intuitively assumed the omitted ns in a backendRef would imply the _parentRef_ namespace or something. Fixed

> note the test is only checking Backend: echo-v1 but we have it in both namespaces, so its untested if we go to the correct namespace?

Not sure that necessarily falls under this test, since I feel like this should just check consumer NS on, non-consumer NS off, but I suppose it should be tested somewhere.

Reply:
Looks like `Backend` is the hostname of the responding Pod so it will just have the Pod's name. So I suppose either we tweak [the echo server](https://github.com/istio/istio/blob/20bb4cad57d6abc5d30d9adb819321d47527909c/pkg/test/echo/response.go#L33) to put the Pod's namespace in the response or we use the k8s API to look up the Pod in the Namespace we expect it to be in using the hostname

Reply:
Oh good point it is just the name. We can easily add namespace. Lets do a followup to do that?